example/operator/15-extension.yaml

```diff
@@ -0,0 +1,25 @@
+apiVersion: operator.gardener.cloud/v1alpha1
+kind: Extension
+metadata:
+  name: local
+#spec:
+#  resources:
+#  - kind: BackupBucket
+#    type: local
+#    primary: true|false
+#    globallyEnabled: true|false # only valid if kind=Extension
+#    reconcileTimeout: 30s # only valid if kind=Extension
+#    lifecycle: {}
+#    workerlessSupported: true|false # only valid if kind=Extension
+#  admissionDeployment:
+#  extensionDeployment:
+#    runtimeDeployment:
+#      helm:
+#        ociRepository: {}
+#        values: {}
+#    gardenDeployment:
+#      helm:
+#        ociRepository: {}
+#        values: {}
```

Comment:
In https://github.com/gardener/gardener/issues/9635, the proposal was to have

```yaml
deployment:
  admission: {}
  extension: {}
```

Why did you deviate from this?

> The reason is that a copy of the extension managing infra resources in the runtime cluster may need to differentiate from a typical seed installation.

Can you explain? Why would an extension (like `provider-aws` or whatever else) need to deploy resources (RBAC, ...) into the virtual cluster? Can you share a scenario/use-case?

Reply:
> Can you explain? Why would an extension (like provider-aws or whatever else) need to deploy resources (RBAC, ...) into the virtual cluster? Can you share a scenario/use-case?

The admission part of each extension which is already neatly separated into 2 charts that target the runtime and garden cluster. The admission needs to deploy it's RBAC to validate/mutate shoot resources, but it is running in the runtime cluster, hence the need for a chart for each of the 2 clusters.

The gardenDeployment section of the `extensionDeployment` refers rather to the `controller{deployment,registration}` part - which is what we actually deploy to the garden cluster. The idea that I tried to convey in the description is that there can be scenarios where the provider-ext running in the operator and the seeds may need to deviate. 
The reasons may be e.g. resource requests needed for the extensions, priority classes and/or other future options, or upgrade/migration scenarios where you want to do a 2 stage upgrade (a stage for the operator version and a stage for the seeds). 

Following that train of though, we need at the very least 2 different values (one for the deployment in operator and one for the `ctrldeploy`. If you think that there will be a need for running the 2 separate versions, then you need the secondary chart (as each version will potentially have it's own chart).
TLDR; (from my PoV):
- if there is a need for something like separate versions of the extension for operator/seeds, then we need 2 charts (can consider defaulting one `OciRepository` from the other even)
- we should consider 2 separate values because otherwise everything in the runtimeCluster would need to be very close to an actual seed and this is a restriction we don't necessarily want. From the hackathon it was the priorityClasses but I can imaging more things may appear
- otherwise, we go back to the original plan of a single chart/value combo (and we try to bridge the gap between runtime clusters and seeds).


> Why did you deviate from this?
Thought I knew better but I digress. Update coming up

Reply:
Aha, makes sense, thanks. Still, I would keep only one `ociRepository` then (i.e., only one version will be used for both runtime and garden deployment).